src/backend/cuda/kernel/medfilt.hpp

```diff
@@ -192,13 +322,35 @@ void medfilt(Param<T> out, CParam<T> in, int w_len, int w_wid)
     dim3 blocks(blk_x*in.dims[2], blk_y*in.dims[3]);
 
     switch(w_len) {
-        case  3: CUDA_LAUNCH((medfilt<T,pad, 3, 3>), blocks, threads, out, in, blk_x, blk_y); break;
-        case  5: CUDA_LAUNCH((medfilt<T,pad, 5, 5>), blocks, threads, out, in, blk_x, blk_y); break;
-        case  7: CUDA_LAUNCH((medfilt<T,pad, 7, 7>), blocks, threads, out, in, blk_x, blk_y); break;
-        case  9: CUDA_LAUNCH((medfilt<T,pad, 9, 9>), blocks, threads, out, in, blk_x, blk_y); break;
-        case 11: CUDA_LAUNCH((medfilt<T,pad,11,11>), blocks, threads, out, in, blk_x, blk_y); break;
-        case 13: CUDA_LAUNCH((medfilt<T,pad,13,13>), blocks, threads, out, in, blk_x, blk_y); break;
-        case 15: CUDA_LAUNCH((medfilt<T,pad,15,15>), blocks, threads, out, in, blk_x, blk_y); break;
+        case  3: CUDA_LAUNCH((medfilt2<T,pad, 3, 3>), blocks, threads, out, in, blk_x, blk_y); break;
+        case  5: CUDA_LAUNCH((medfilt2<T,pad, 5, 5>), blocks, threads, out, in, blk_x, blk_y); break;
+        case  7: CUDA_LAUNCH((medfilt2<T,pad, 7, 7>), blocks, threads, out, in, blk_x, blk_y); break;
+        case  9: CUDA_LAUNCH((medfilt2<T,pad, 9, 9>), blocks, threads, out, in, blk_x, blk_y); break;
+        case 11: CUDA_LAUNCH((medfilt2<T,pad,11,11>), blocks, threads, out, in, blk_x, blk_y); break;
+        case 13: CUDA_LAUNCH((medfilt2<T,pad,13,13>), blocks, threads, out, in, blk_x, blk_y); break;
+        case 15: CUDA_LAUNCH((medfilt2<T,pad,15,15>), blocks, threads, out, in, blk_x, blk_y); break;
+    }
+
+    POST_LAUNCH_CHECK();
+}
+
+template<typename T, af_border_type pad>
+void medfilt1(Param<T> out, CParam<T> in, int w_wid)
+{
+    const dim3 threads(THREADS_X);
+
+    int blk_x = divup(in.dims[0], threads.x);
+
+    dim3 blocks(blk_x*in.dims[1], in.dims[2], in.dims[3] );
+
+    switch(w_wid) {
+        case  3: CUDA_LAUNCH((medfilt1<T,pad, 3>), blocks, threads, out, in, blk_x); break;
+        case  5: CUDA_LAUNCH((medfilt1<T,pad, 5>), blocks, threads, out, in, blk_x); break;
+        case  7: CUDA_LAUNCH((medfilt1<T,pad, 7>), blocks, threads, out, in, blk_x); break;
+        case  9: CUDA_LAUNCH((medfilt1<T,pad, 9>), blocks, threads, out, in, blk_x); break;
+        case 11: CUDA_LAUNCH((medfilt1<T,pad,11>), blocks, threads, out, in, blk_x); break;
+        case 13: CUDA_LAUNCH((medfilt1<T,pad,13>), blocks, threads, out, in, blk_x); break;
+        case 15: CUDA_LAUNCH((medfilt1<T,pad,15>), blocks, threads, out, in, blk_x); break;
```

Comment:
1D stuff median filter need not be limited to 15. Make it bigger (and remove templates)